moveit_core/planning_scene/include/moveit/planning_scene/planning_scene.h

```diff
@@ -372,114 +365,80 @@ class MOVEIT_PLANNING_SCENE_EXPORT PlanningScene : public std::enable_shared_fro
       a non-const \e robot_state and updates its link transforms if needed. */
   void checkCollision(const collision_detection::CollisionRequest& req, collision_detection::CollisionResult& res,
                       moveit::core::RobotState& robot_state,
-                      const collision_detection::AllowedCollisionMatrix& acm) const
-  {
-    robot_state.updateCollisionBodyTransforms();
-    checkCollision(req, res, static_cast<const moveit::core::RobotState&>(robot_state), acm);
-  }
+                      const collision_detection::AllowedCollisionMatrix& acm) const;
 
   /** \brief Check whether a specified state (\e robot_state) is in collision, with respect to a given
-      allowed collision matrix (\e acm). */
+      allowed collision matrix (\e acm). If validate_transforms is set true, the given robot state is checked for dirty transforms. */
```

Comment:
See my other comment, I find this API confusing since `validate_transforms` is not making this function fail or succeed.